main.c

```diff
@@ -275,11 +275,23 @@ static void elf_boot(const struct boot_params params)
         const uint8_t *current_elf_segment_in_memory =
             elf_addr_in_memory + current_program_header->offset;
 
-        printf(
-            "Loading ELF segment (type 0x%x, offset 0x%x, vaddr 0x%x, paddr 0x%x, filesize 0x%x, memsize 0x%x)\n",
-            current_program_header->type, current_program_header->offset,
-            current_program_header->vaddr, current_program_header->paddr,
-            current_program_header->filesize, current_program_header->memsize);
+        if (current_program_header->type != 1 /* LOAD segment */) {
+            printf(
+                "Skipping Non-LOAD ELF segment (type 0x%x, offset 0x%x, vaddr 0x%x, paddr 0x%x, filesize 0x%x, memsize 0x%x)\n",
+                current_program_header->type, current_program_header->offset,
+                current_program_header->vaddr, current_program_header->paddr,
+                current_program_header->filesize, current_program_header->memsize);
+            continue;
+        } else {
+            printf(
+                "Loading ELF segment (type 0x%x, offset 0x%x, vaddr 0x%x, paddr 0x%x, filesize 0x%x, memsize 0x%x)\n",
+                current_program_header->type, current_program_header->offset,
+                current_program_header->vaddr, current_program_header->paddr,
+                current_program_header->filesize, current_program_header->memsize);
+        }
```

Comment:
nit: this would also be fine without the `else` branch and cause less code churn.